alf/networks/preprocessor_networks.py

```diff
@@ -83,19 +83,21 @@ def _get_preprocessed_spec(preproc, spec):
             def _return_or_copy_preprocessor(preproc):
                 if preproc is None:
                     # allow None as a placeholder in the nest
-                    preproc = math_ops.identity
-                elif isinstance(preproc, InputPreprocessor):
+                    return math_ops.identity
+                elif isinstance(preproc, Network):
                     preproc = preproc.copy()
                     self._input_preprocessor_modules.append(preproc)
-                return preproc
+                    return lambda x: preproc(x)[0]
+                else:
```

Comment:
Also here we should consider preproc as a nn.Module

Reply:
It is not a generic nn.Module, still Network.

Reply:
What if 
```python
input_preprocessors={
    "camera": @ResnetEncodingNetwork(
        input_tensor_spec=alf.nest.get_field(%observation_spec, "camera"), output_size=%encoding_dim, output_activation=alf.math.identity),
    "radar": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%radar_spec), %encoding_dim, use_bias=False)),
    "collision": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%collision_spec), %encoding_dim, use_bias=False)),
    "gnss": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%goal_spec), %encoding_dim, use_bias=False)),
    "imu": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%imu_spec), %encoding_dim, use_bias=False)),
    "goal": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%goal_spec), %encoding_dim, use_bias=False)),
    "speed": torch.nn.Sequential(alf.layers.Reshape([-1]), alf.layers.FC(@flattened_size(%speed_spec), %encoding_dim, use_bias=False)),
}
```

And then I copy the network? The layers are still shared.

Reply:
I also added a Network call SequentialNetwork which accept a list of torch layers. In PreprocessNetwork, if the preprocessor is not a Network, I construct a SequentialNetork from it.

Reply:
That sounds great. It will bridge the gap between nn.module and Network.